jalib/jsocket.cpp

```diff
@@ -152,25 +154,52 @@ jalib::JSockAddr::JSockAddr(const char *hostname /* == NULL*/,
 
 jalib::JSocket::JSocket()
 {
-  _sockfd = jalib::socket(AF_INET, SOCK_STREAM, 0);
+
+	if( jalib::JSocket::isFileSocketDefined() ) {
+
+		memset( &FILE_SOCKET_ADDR, 0, sizeof(struct sockaddr_un) );
+		FILE_SOCKET_ADDR.sun_family = AF_UNIX;
+		strncpy( FILE_SOCKET_ADDR.sun_path, getenv(ENV_VAR_DMTCP_FILE_SOCKET_PATH), sizeof(FILE_SOCKET_ADDR.sun_path) - 1 );
+		_sockfd = jalib::socket(AF_UNIX, SOCK_STREAM, 0);
+	} else {
+    _sockfd = jalib::socket(AF_INET, SOCK_STREAM, 0);
+	}
+}
+
+bool jalib::JSocket::isFileSocketDefined() {
+  bool isDefined = false;
+  if ( getenv( ENV_VAR_DMTCP_FILE_SOCKET_PATH ) != NULL ) {
+        isDefined = true;
+  }
+  return isDefined;
+}
+
+char * jalib::JSocket::getFileSocketPath() {
+  return getenv( ENV_VAR_DMTCP_FILE_SOCKET_PATH ); // requires #include "../src/constants.h"
 }
 
+
 bool
 jalib::JSocket::connect(const JSockAddr &addr, int port)
 {
   bool ret = false;
 
-  // jalib::JSockAddr::JSockAddr used -2 to poison port (invalid host)
-  if (addr._addr->sin_port == (unsigned short)-2) {
-    return false;
-  }
-  for (unsigned int i = 0; i < addr._count; i++) {
-    ret = JSocket::connect((sockaddr *)(addr._addr + i),
-                           sizeof(addr._addr[0]),
-                           port);
-    if (ret || errno != ECONNREFUSED) {
-      break;
-    }
+  if( !jalib::JSocket::isFileSocketDefined() ) {
+		// jalib::JSockAddr::JSockAddr used -2 to poison port (invalid host)
+		if (addr._addr->sin_port == (unsigned short)-2) {
+			return false;
+		}
+		for (unsigned int i = 0; i < addr._count; i++) {
+			ret = JSocket::connect((sockaddr *)(addr._addr + i),
+														 sizeof(addr._addr[0]),
+														 port);
+			if (ret || errno != ECONNREFUSED) {
+				break;
+			}
+		}
+		return ret;
+  } else {
+  	return ::connect(_sockfd, (const struct sockaddr *) &FILE_SOCKET_ADDR, sizeof(struct sockaddr_un));
```

Comment:
For blocks like these (here and below), it might be better to have something like:
```if (jalib::JSocket::isFileSocketDefined()) {
  return ::connect(_sockfd, (const struct sockaddr *) &FILE_SOCKET_ADDR, sizeof(struct sockaddr_un));
}```

That way the rest of the code block stays the same.